### PR TITLE
Update ShadowContentResolver#isBundleEqual() to compare Bundles that contain Bundles.

### PR DESCRIPTION
**Pull request by saguinav**
From discussion https://github.com/robolectric/robolectric/issues/2175

### CODE REVIEW

**Comment by jongerrish** [comment]
Can we add a test for this?

File: robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowContentResolver.java [hunk 1]
```diff
@@ -588,12 +588,25 @@ private static boolean isBundleEqual(Bundle bundle1, Bundle bundle2) {
...
+        }
+      } else if (!value1.equals(value2)) {
         return false;
       }
     }
     return true;
   }
+  
+  private static boolean isBundle(Object expected) {
+      return expected != null && expected instanceof Bundle;
```
Comment (dhaag23) [inline robolectric-shadows/shadows-core/src/main/java/org/robolectric/shadows/ShadowContentResolver.java]: `expected != null` is redundant. All you need is `expected instanceof Bundle`.

**Comment by utzcoz** [comment]
Hi @saguinav , could you help to rebase this PR based on latest mater and rework this PR? Thanks.

**Comment by utzcoz** [comment]
@saguinav Thanks for your contribution. If you think this issue still exists and you want to improve it, please send a new PR based on the current code base, and this PR is too old, so I will close this PR.

**Pull request closed without merging**